doc/design/quic-programming-guide.md

```diff
@@ -0,0 +1,317 @@
+
+
+# Overview
+
+OWT QUIC SDK and Client SDKs provides the APIs for enabling QUIC Transport for reliable data transmission over QUIC protocol with OWT Conference Server.
+
+# Scope
+
+This document describes the programming models and API interfaces for following usage scenarios.
+
+- Deploying a QUIC conference server that is capable of forwarding data over QUIC channel.
+- Implementing client application that is capable of sending data to QUIC server or receiving data from it.
+
+Description of the details of QUIC transport is outside the scope of this document.
+
+# Related Repos
+
+Below are the repo locations of current SDK and server implementations:
+
+- OWT QUIC C++ SDK: [https://gitlab.devtools.intel.com/open-webrtc-toolkit/owt-deps-quic](https://gitlab.devtools.intel.com/open-webrtc-toolkit/owt-sdk-quic) This is the C++ implementation of Server-side and Client-side SDK, and is the base of enabling QUIC agent on server, and QUIC conference SDK on client.
+- OWT Conference server: [https://github.com/open-webrtc-toolkit/owt-server](https://github.com/open-webrtc-toolkit/owt-server/pull/113). This is the server repo that supports forwarding of QUIC streams.
+- OWT JavaScript SDK: [https://github.com/open-webrtc-toolkit/owt-client-javascript](https://github.com/open-webrtc-toolkit/owt-client-javascript) This is the repo for enabling QUIC client on browser.
+- OWT Native SDK: [https://github.com/open-webrtc-toolkit/owt-client-native](https://github.com/open-webrtc-toolkit/owt-client-native) This is the client SDK repo for enabling QUIC support using OWT conferencing API.
+
+# Architecture
+
+The topology of components is shown in below diagram:
+
+![plot](./pics/quic_block_diagram.png)
+
+There are a few components involved and their relationships with streaming using QuicTransport are described as below:
+
+## OWT QUIC C++ SDK
+
+This is the foundation of QUIC implementation in OWT. It provides the APIs to create QUIC transport clients and server forming a C/S architecture. Basically you can directly build your QUICTransport applications using the QUIC C++ SDK client API if you're not using the OWT native SDK.
+
+## OWT Native SDK for Conference
+
+The OWT conference SDK built upon OWT QUIC C++ SDK client API. It is used in combination with OWT QUIC conference server when you rely on OWT signaling protocol for QUIC connection setup as well as stream forwarding.
+
+## OWT QUIC Conference Server
+
+The OWT QUIC conference server implements the signaling protocol for QUIC connection setup, as well as QUIC stream forwarding.
+
+## OWT QUIC JavaScript SDK
+
+Used together with OWT QUIC conference server, to build Web-based QUIC applications. Useful when the QUIC streaming application is implemented using Web API.
+
+# How to build OWT QUIC C++ SDK
+
+Below are the steps for building the QUIC SDK from [https://gitlab.devtools.intel.com/open-webrtc-toolkit/owt-deps-quic](https://gitlab.devtools.intel.com/open-webrtc-toolkit/owt-sdk-quic) .
```

Comment:
Please update the URL to GitHub repo.

Reply:
done by referring to doc on related repo.